src/main/kotlin/io/omnitrade/client/OmnitradeClient.kt

```diff
@@ -0,0 +1,407 @@
+      "id" to id.toString()
+    )
+
+    options["signature"] = this.auth.signedParams(
+      HttpMethod.POST.toString(),
+      "$API_VERSION/order/delete",
+      options
+    )
+
+    return omnitradeRepository.deleteOrder(options).withExceptionOnError()
+  }
+
+  fun clearAllOrders(market: Market? = null, side: OrderSide? = null): CompletableFuture<List<Order>> {
+    checkKeys()
+
+    val options = mutableMapOf<String, String>()
+    market?.let { options["market"] = it.id }
+    side?.let { options["side"] = it.side }
+
+    options["signature"] = this.auth.signedParams(
+      HttpMethod.POST.toString(),
+      "$API_VERSION/orders/clear",
+      options
+    )
+
+    return omnitradeRepository.clearAllOrders(options).withExceptionOnError()
+  }
+
+  fun clearBottomOrder(market: Market? = null, side: OrderSide? = null, depth: OrderDepth? = null): CompletableFuture<List<Order>> {
+    checkKeys()
+
+    val options = mutableMapOf<String, String>()
+    market?.let { options["market"] = it.id }
+    side?.let { options["side"] = it.side }
+    depth?.let { options["depth"] = it.depth }
+
+    options["signature"] = this.auth.signedParams(
+      HttpMethod.POST.toString(),
+      "$API_VERSION/orders/clear_bottom",
+      options
+    )
+
+    return omnitradeRepository.clearBottomOrder(options).withExceptionOnError()
+  }
+
+  fun getMyTrades(market: Market, tradeIdSince: Int? = null, tradeIdUntil: Int? = null, timestamp: Long? = null, limit: Int? = null, orderBy: OrderBy): CompletableFuture<List<Trade>> {
+    checkKeys()
+
+    val options = mutableMapOf(
+      "market" to market.id,
+      "order_by" to orderBy.order
+    )
+
+    tradeIdSince?.let { options["since"] = it.toString() }
+    tradeIdUntil?.let { options["until"] = it.toString() }
+    timestamp?.let { options["timestamp"] = it.toString() }
+    limit?.let { options["limit"] = it.toString() }
+
+    options["signature"] = this.auth.signedParams(
+      HttpMethod.GET.toString(),
+      "$API_VERSION/trades/my",
+      options
+    )
+
+    return omnitradeRepository.getMyTrades(options).withExceptionOnError()
+  }
+
+  fun setupKeys(accessKey: String, secretKey: String): OmnitradeClient {
+    if (accessKey.isEmpty() || secretKey.isEmpty()) {
+      throw OmnitradeNoAuthKeysException()
+    }
+
+    this.accessKey = accessKey
+    this.secretKey = secretKey
+    return this
+  }
+
+  private fun checkKeys() {
+    if (!auth.hasKeys()) {
+      throw OmnitradeNoAuthKeysException()
+    }
+  }
+
+  private fun <T> handleResponse(response: Response<T>): T? {
+    if (response.isSuccessful) {
+      return response.body()
+    } else {
+      throw OmnitradeRequestException(response.code().toString(), response.message())
+    }
+  }
+
+  private fun <T> CompletableFuture<T>.withExceptionOnError(): CompletableFuture<T> {
+    return this.exceptionally { ex ->
+      when(ex) {
+        is HttpException -> {
+          if (ex.code() == 404) null
```

Comment:
Is there some constant from the Kotlin core that already encapsulates this number? Something like `Http.Status.NOT_FOUND` or something like that?

Reply:
Actually, there is! `java.net.HttpURLConnection.HTTP_NOT_FOUND` will do the job ;)